src/common/crash_dump/dump.cpp

```diff
@@ -11,7 +11,7 @@ static constexpr uint32_t dump_offset = w25x_dump_start_address;
 static constexpr uint16_t dump_buff_size = 0x100;
 static constexpr uint32_t dump_xflash_size = DUMP_RAM_SIZE + DUMP_CCRAM_SIZE;
 
-static_assert(dump_xflash_size <= w25x_pp_start_address, "Dump overflows reserved space.");
+static_assert(dump_xflash_size <= w25x_error_start_adress, "Dump overflows reserved space.");
```

Comment:
Suggestion:
Add check that error is not overflowing reserved space:
e.g. `static_assert(sizeof(dumpmessage_t) <= (w25x_pp_start_address - w25x_error_start_adress), "Error message overflows reserved space.");`

Reply:
Good point, added in squashed commit